server/graphql/v2/mutation/AgreementMutations.ts

```diff
@@ -47,16 +52,19 @@ export default {
         throwIfMissing: true,
       });
 
+      const account = await fetchAccountWithReference(args.account, {
+        throwIfMissing: true,
+      });
+
       if (!req.remoteUser.isAdminOfCollective(host)) {
         throw new Unauthorized('Only host admins can create agreements');
+      } else if (account.HostCollectiveId !== host.id) {
+        // We're not checking `isActive` as it should be possible to create agreements for accounts not approved yet
+        throw new Unauthorized(`Account ${account.name} is not currently hosted by ${host.name}`);
```

Comment:
@hdiniz Related to our conversation to only let users add agreements on accounts they're currently hosting.